cmd/wallet-sdk-gomobile/docs/packages.md

```diff
@@ -0,0 +1,400 @@
+and compare how something in Go gets converted to Java (for Android) and Objective-C (for iOS).
+
+These examples are not meant to be full code examples of how you actually use the API.
+Rather, these examples are just to help you understand the naming patterns, which should help you in figuring out
+the equivalent names for various parts of the API across platforms.
+
+The left-most side in the tables shows Go source code snippets, rather than how they would look when imported in another
+file. The middle and right-most side show how the code looks when importing that same functionality in via the bindings.
+The reason for comparing between Go and the bindings like this is that the Go code is not intended to be imported
+directly - the mobile wrappers exist only to make the Go code compatible with the `gomobile` tool, so seeing how it
+would look imported is not so useful. However, the way you import from the generated mobile bindings is important,
+and so that's why we show that here. The Go source code is really just meant to help demonstrate the naming patterns
+to the reader (as the names used for packages/modules in the bindings are generated automatically by `gomobile` based
+on the Go names).
+
+### API
+
+This package contains interface definitions and types that are used in multiple places across the API as a whole.
+
+<table>
+<tr>
+<td> Go Source Code </td> <td> Java (as called from Kotlin) </td> <td> Obj-C (as called from Swift) </td>
+</tr>
+<tr>
+<td> 
+
+```
+package api
+
+type CredentialReader interface {
+	Get(...) (*JSONObject, error)
+	GetAll() (*JSONArray, error)
+}
+```
+
+</td>
+<td>
+
+```kotlin
+import dev.trustbloc.wallet.sdk.api
+
+api.CredentialReader
+```
+
+</td>
+<td>
+
+```swift
+import Walletsdk
+
+ApiCredentialReader
+```
+
+</td>
+</tr>
+</table>
+
+### Credential Querying
+
+<table>
+<tr>
+<td> Go Source Code </td> <td> Java (as called from Kotlin) </td> <td> Obj-C (as called from Swift) </td>
+</tr>
+<tr>
+<td> 
+
+```
+package credential
+
+func NewInquirer(
+	...
+) *Inquirer {
+	...
+}
+```
+
+</td>
+<td>
+
+```kotlin
+import dev.trustbloc.wallet.sdk.credential
+
+val inquirer = credential.newInquirer(...)
+
+inquirer.query(...)
+```
+
+</td>
+<td>
+
+```swift
+import Walletsdk
+
+let inquirer = CredentialNewInquirer(...)
+
+inquirer.query(...)
+```
+
+</td>
+</tr>
+</table>
+
+### DID Creator and Resolver
+
+<table>
+<tr>
+<td> Go Source Code </td> <td> Java (as called from Kotlin) </td> <td> Obj-C (as called from Swift) </td>
+</tr>
+<tr>
+<td> 
+
+```
+package did
+
+func NewCreatorWithKeyWriter(
+	...
+) (*Creator, error) {
+    ...
+}
+
+func NewResolver() *Resolver {
+	...
+}
+
+```
+
+</td>
+<td>
+
+```kotlin
+import dev.trustbloc.wallet.sdk.did
+
+did.newCreatorWithKeyWriter(...)
+did.newResolver()
+```
+
+</td>
+<td>
+
+```swift
+import Walletsdk
+
+DidNewCreatorWithKeyWriter(...)
+DidNewResolver()
+```
+
+</td>
+</tr>
+</table>
+
+### In-Memory Credential DB
```

Comment:
(in Aries, we had the provider vs store distinction because a provider could create multiple stores, however here we don't need the multiple level structure)